ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -158,6 +163,10 @@ func (scbuilder *SiteConfigBuilder) getClusterCRs(clusterId int, siteConfigTemp
 						clusterCRs = append(clusterCRs, instantiatedCR)
 					}
 				} else {
+					if kind == "BareMetalHost" {
```

Comment:
Why is this if inside the else instead of else if kind == "BareMetalHost" like HostFirmwareSettings and NMStateConfig?

Reply:
Because 'clusterCRs = append(clusterCRs, instantiatedCR)' is common under else, otherwise it is called under else if and else.